internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -188,50 +187,107 @@ func resourceMonitorDiagnosticSetting() *pluginsdk.Resource {
+						Optional: true,
+						Default:  true,
+					},
+
+					"retention_policy": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+								},
+
+								"days": {
+									Type:         pluginsdk.TypeInt,
+									Optional:     true,
+									ValidateFunc: validation.IntAtLeast(0),
+								},
+							},
+						},
+					},
+				},
+			},
+			Set: resourceMonitorDiagnosticLogSettingHash,
+		}
+	}
+
+	return resource
 }
 
-func resourceMonitorDiagnosticSettingCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+func resourceMonitorDiagnosticSettingCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Monitor.DiagnosticSettingsClient
-	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 	log.Printf("[INFO] preparing arguments for Azure ARM Diagnostic Settings.")
 
-	name := d.Get("name").(string)
-	actualResourceId := d.Get("target_resource_id").(string)
-	diagnosticSettingId := diagnosticsettings.NewScopedDiagnosticSettingID(actualResourceId, name)
+	id := diagnosticsettings.NewScopedDiagnosticSettingID(d.Get("target_resource_id").(string), d.Get("name").(string))
 
-	if d.IsNewResource() {
-		existing, err := client.Get(ctx, diagnosticSettingId)
-		if err != nil {
-			if !response.WasNotFound(existing.HttpResponse) {
-				return fmt.Errorf("checking for presence of existing Monitor Diagnostic Setting %q for Resource %q: %s", diagnosticSettingId.Name, diagnosticSettingId.ResourceUri, err)
-			}
+	existing, err := client.Get(ctx, id)
+	if err != nil {
+		if !response.WasNotFound(existing.HttpResponse) {
+			return fmt.Errorf("checking for presence of existing Monitor Diagnostic Setting %q for Resource %q: %s", id.Name, id.ResourceUri, err)
 		}
+	}
 
-		if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
-			return tf.ImportAsExistsError("azurerm_monitor_diagnostic_setting", *existing.Model.Id)
-		}
+	if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+		return tf.ImportAsExistsError("azurerm_monitor_diagnostic_setting", *existing.Model.Id)
 	}
 
-	logsRaw := d.Get("log").(*pluginsdk.Set).List()
-	logs := expandMonitorDiagnosticsSettingsLogs(logsRaw)
 	metricsRaw := d.Get("metric").(*pluginsdk.Set).List()
 	metrics := expandMonitorDiagnosticsSettingsMetrics(metricsRaw)
 
-	// if no blocks are specified  the API "creates" but 404's on Read
-	if len(logs) == 0 && len(metrics) == 0 {
-		return fmt.Errorf("At least one `log` or `metric` block must be specified")
+	var logs []diagnosticsettings.LogSettings
+	hasEnabledLogs := false
+	if !features.FourPointOhBeta() {
+		logsRaw := d.Get("log").(*pluginsdk.Set).List()
```

Comment:
since this field is Computed and will get a default value, do we need to use `GetOk` here?